addons/mail/static/src/model/model_manager.js

```diff
@@ -708,6 +709,15 @@ class ModelManager {
         for (const record of this._updatedRecords) {
             record.__state++;
         }
+
+        for (const record of this._updatedRecords) {
+            for (const required of record.constructor.__requiredFieldsList) {
+                if (record[required] === undefined) {
+                    throw Error(`Field ${required.fieldName} of ${record.localId} is required.`);
+                }
+            }
+        }
```

Comment:
would put 1-line comment above block of code, e.g. `// verify required fields are set`